### ISSUE TITLE
method parameter incompatible, expect syntax error

### ISSUE THREAD

**Issue by larshp** [abaplint/abaplint#2380]
passing string to tab is okay, its just tab to string that is not reported, huh,

```abap
CLASS lcl DEFINITION.
  PUBLIC SECTION.
    METHODS method1 IMPORTING str TYPE string.
ENDCLASS.
CLASS lcl IMPLEMENTATION.
  METHOD method1.
    DATA tab TYPE STANDARD TABLE OF string WITH DEFAULT KEY.
    method1( tab ).
  ENDMETHOD.
ENDCLASS.
```
